Src/FluentAssertions/Equivalency/EqualityComparerEquivalencyStep.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class EqualityComparerEquivalencyStep<T> : IEquivalencyStep
+    {
+        private readonly IEqualityComparer<T> comparer;
+
+        public EqualityComparerEquivalencyStep(IEqualityComparer<T> comparer)
+        {
+            this.comparer = comparer;
+        }
+
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return context.RuntimeType == typeof(T);
+        }
+
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
+        {
+            Execute.Assertion
+                .BecauseOf(context.Because, context.BecauseArgs)
+                .ForCondition(comparer.Equals((T)context.Subject, (T)context.Expectation))
+                .FailWith("Expected {context:object} to be equal according to {0}{because}, but it was not.", comparer.ToString());
```

Comment:
Does the failure message contain both the subject and expectation?

Reply:
Now it does.